python/cudf/cudf/_lib/strings/json.pyx

```diff
@@ -27,16 +31,33 @@ def get_json_object(Column col, object py_json_path, bool allow_single_quotes=Fa
         json_path.get_raw_ptr()
     )
 
-    cdef get_json_object_options options
-    options.set_allow_single_quotes(allow_single_quotes)
-    options.set_strip_quotes_from_single_strings(strip_quotes_from_single_strings)
-    options.set_missing_fields_as_nulls(missing_fields_as_nulls)
-
     with nogil:
         c_result = move(cpp_get_json_object(
             col_view,
             scalar_json_path[0],
-            options,
+            options.options,
         ))
 
     return Column.from_unique_ptr(move(c_result))
+
+
+cdef class GetJsonObjectOptions:
+    cdef get_json_object_options options
+
+    def get_allow_single_quotes(self):
```

Comment:
Instead of getters and setters, would it be preferable to add a constructor (`__init__`) method and `property` methods to this class?

Reply:
I debated this as well — but it gets a bit awkward for large classes with many options. A constructor could have been added at the C++ level instead except it runs into the awkward limitations of keyword arguments in C++. We used a fluent builder pattern for similar options structures with parquet / other I/O features. I didn’t like any of my alternative ideas but it might be fine to have a constructor in Python and not C++…

Reply:
Yeah large constructors are fine as long as we have reasonable defaults for parameters. Are there good examples of the equivalent of the builder pattern in Python? E.g., in the standard library?

Reply:
I would be fine with a constructor in Python. I recall recommending this somewhere else as well. Depending on whether this really needs to be a cdef class, we could use a Python dataclass instead. If having it be a cdef class is important, maybe just make all of the `__init__` parameters keyword-only arguments so that it makes it really obvious what options are being set (`options = GetJsonObjectOptions(opt1=..., opt2=...)`).

Reply:
Agree, let's make the ctor arguments keyword-only.